sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -952,9 +942,25 @@ export class MessageReceiver extends LinkEntity {
           connectionHost: this._context.namespace.config.host,
           delayInSeconds: 15
         };
-        if (!this.wasCloseInitiated) {
-          await retry<void>(config);
-        }
+        await retry<void>(config);
+      } else if (causedByDisconnect) {
+        log.error(
+          "[%s] Something went wrong and the connection disconnected. Receiver '%s' with address '%s' encountered error: %O",
+          connectionId,
+          this.name,
+          this.address,
+          translatedError
+        );
+        // Not retrying, throw the error so it gets logged and forwarded to user's error handler.
+        throw translatedError;
+      } else {
+        log.error(
+          "[%s] Something went wrong. Receiver '%s' with address '%s' encountered error: %O",
```

Comment:
To be in sync with my other suggestion:
```suggestion
          "[%s] Encountered a non retryable error on the receiver. Cannot recover receiver '%s' with address '%s' encountered error: %O",
```